### ISSUE TITLE
Update to new YAML config file format in MongoDB >= 2.6

### ISSUE THREAD

**Issue by hackel** [saltstack-formulas/mongodb-formula#31]
I don't have time to do this now, but I wanted to make a note of it.  The current mongodb.conf file uses the old Mongo 2.4 config format.  The newer format should be much easier, especially since we can have a 1:1 mapping with our pillar file.  Here's the default file for reference:
```yaml
# mongod.conf

# for documentation of all options, see:
#   http://docs.mongodb.org/manual/reference/configuration-options/

# Where and how to store data.
storage:
  dbPath: /var/lib/mongodb
  journal:
    enabled: true
#  engine:
#  mmapv1:
#  wiredTiger:

# where to write logging data.
systemLog:
  destination: file
  logAppend: true
  path: /var/log/mongodb/mongod.log

# network interfaces
net:
  port: 27017
  bindIp: 0.0.0.0


#processManagement:

#security:

#operationProfiling:

#replication:

#sharding:

## Enterprise-Only Options:

#auditLog:

#snmp:
```

Note that this also switches to using syslogd, which seems like a good thing.

**Comment by gravyboat** [comment]
Is there going to be a problem with the new conf file layout for older mongo versions?

**Comment by hackel** [comment]
Yeah, we'd have to include templates for both if we want to support them.  No clue what the version distribution looks like, but 2.6 was released nearly 2 years ago...

Actually, it looks like Ubuntu 14.04 only came with 2.4.9, so I suppose the old format will still have to be supported.

**Comment by sbreidba** [comment]
I have a patch for mongodb.conf.jinja for this. I needed to implement it when I used the formula defaults as those grabbed a 3.x version of mongo. It looks like this:

    {% if 'mongod_settings' in mdb -%}
    {{ mdb.mongod_settings | yaml(false) }}
    {% else -%}
    <existing code for non-YAML config file...>

It's fairly convenient as @hackel says, because we can just dump a subset of the pillar into the config file - no need to do our own mappings case-by-case. I made it a sub-key in the pillar so that we could include metadata as needed (`version` is an obvious candidate).

However, there are other issues that keep this from being production-worthy. For example, the name of the config file changed as did the service name, and I haven't touched the mongos config since I have no experience with it.

Is this worth pursuing? I'm glad to submit what I've done for review. 

(Related: do we even want/need to support 2.4? Outside of what's mentioned above, the only thing I needed to do to make the formula work with 3.2 was to import the apt key as per the mongo documentation.)

**Comment by genuss** [comment]
I think this simple can be accepted, because it will resolve 95% of issues.
Yes, we won't have default values properly configured, but this isn't critical (IMHO).

**Comment by sbreidba** [comment]
I'll do a bit of clean up and submit a patch for review shortly.

**Comment by gravyboat** [comment]
This is resolved over in #35.
